src/main/java/kitchenpos/menu/domain/Menu.java

```diff
@@ -16,42 +17,37 @@ public class Menu {
     @Embedded
     private Price price;
 
-    @ManyToOne(fetch = FetchType.LAZY)
-    private MenuGroup menuGroup;
+    private Long menuGroupId;
 
     @Embedded
     private MenuProducts menuProducts = new MenuProducts();
 
     protected Menu() {
     }
 
-    public Menu(Long id, String name, Price price, MenuGroup menuGroup) {
+    public Menu(Long id, String name, Price price, Long menuGroupId) {
         this.id = id;
         this.name = name;
         this.price = price;
-        this.menuGroup = menuGroup;
+        this.menuGroupId = menuGroupId;
     }
 
-    public Menu(String name, Price price, MenuGroup menuGroup) {
+    public Menu(String name, Price price, Long menuGroupId, MenuProducts menuProducts) {
         this.name = name;
         this.price = price;
-        this.menuGroup = menuGroup;
+        this.menuGroupId = menuGroupId;
+        this.menuProducts = menuProducts;
     }
 
 
-    private static void validCheckMeuProductPrice(BigDecimal sumPrice, Price price) {
+    public void validCheckMeuProductPrice() {
+        BigDecimal sumPrice = this.menuProducts.getSumPrice();
```

Comment:
```suggestion
        Price sumPrice = this.menuProducts.getSumPrice();
```
메뉴 상품의 합도 Price로 봐도 되지않을까요? 🤔

또한 Price와 Price를 비교할때도 `price.getPrice()` 와 같이 getter를 통해 값을 꺼내와서 처리하기보다는
`price.isGreaterThan(sumPrice);` 와 같이 사용하는게 더 나을거같아요.